default/scripting/buildings/TRANSLATOR.focs.py

```diff
@@ -0,0 +1,76 @@
+from buildings.buildings import SPECIES_LIKES_OR_DISLIKES_BUILDING_STABILITY_EFFECTS
+from focs._effects import (
+    Capital,
+    Contains,
+    CountUnique,
+    EffectsGroup,
+    Focus,
+    HasSpecies,
+    IsBuilding,
+    LocalCandidate,
+    NamedReal,
+    Object,
+    OwnedBy,
+    Planet,
+    SetEmpireMeter,
+    SetTargetInfluence,
+    Source,
+    Statistic,
+    TargetPopulation,
+    Unowned,
+    Value,
+)
+from macros.base_prod import BUILDING_COST_MULTIPLIER
+from macros.enqueue import ENQUEUE_BUILD_ONE_PER_PLANET
+from macros.priorities import (
+    TARGET_AFTER_2ND_SCALING_PRIORITY,
+)
+
+try:
+    from focs._buildings import *
+except ModuleNotFoundError:
+    pass
+
+BuildingType(  # type: ignore[reportUnboundVariable]
+    name="BLD_TRANSLATOR",
+    description="BLD_TRANSLATOR_DESC",
+    buildcost=320 * BUILDING_COST_MULTIPLIER,
+    buildtime=10,
+    location=(
+        Planet()
+        & OwnedBy(empire=Source.Owner)
+        & ~Contains(IsBuilding(name=["BLD_TRANSLATOR"]))
+        & TargetPopulation(low=1)
+    ),
+    enqueuelocation=ENQUEUE_BUILD_ONE_PER_PLANET,
+    effectsgroups=[
+        *SPECIES_LIKES_OR_DISLIKES_BUILDING_STABILITY_EFFECTS,
+        EffectsGroup(
+            scope=(Capital & OwnedBy(empire=Source.Owner)),
+            activation=(Unowned),
+            stackinggroup="TRANSLATOR_POLICY_SLOT_STACK",
+            effects=[SetEmpireMeter(empire=Source.Owner, meter="SOCIAL_CATEGORY_NUM_POLICY_SLOTS", value=Value + 1)],
+        ),
+        EffectsGroup(
+            scope=(Object(id=Source.PlanetID) & Planet() & Unowned & Focus(type=["FOCUS_INFLUENCE"])),
+            activation=(Unowned),
+            priority=TARGET_AFTER_2ND_SCALING_PRIORITY,
+            effects=[
+                SetTargetInfluence(
+                    value=Value
+                    + NamedReal(name="TRANSLATOR_PER_SPECIES_INFLUENCE", value=0.5)
+                    * (
+                        Statistic(
+                            float,
```

Comment:
Now should be possible to declare both types `float` and `str`.

Reply:
Usually, the statistic value is the same for the processed and returned argument.  

But this case is special.  We process `string` and return `int`.

Code fails with:   

```
14:51:54.790679 {0x0000277c} [error] ai : PythonCommon.cpp:145 : TypeError: unsupported operand type(s) for *: 'ValueRefDouble' and 'ValueRefString'
14:51:54.790679 {0x0000277c} [error] ai : PythonParser.cpp:410 : Unable to parse data file j:/projects/freeorion/default\scripting/buildings\TRANSLATOR.focs.py
```

Reply:
Yes, so it should be declared as `Statistic(float, str, CountUnique, ...)`

Reply:
Just to clarify, this already should work or is an update of C++ required?



```python
Statistic(
                            float, str,
                            CountUnique,
                            value=LocalCandidate.Species,
                            condition=Planet() & HasSpecies() & OwnedBy(empire=Source.Owner),
                        )
```

```
16:06:49.590654 {0x00001910} [error] ai : PythonCommon.cpp:145 :   File "<string>", line 63, in <module>
16:06:49.590654 {0x00001910} [error] ai : PythonCommon.cpp:145 : TypeError: No registered converter was able to produce a C++ rvalue of type struct enum_wrapper<enum ValueRef::StatisticType> from this Python object of type type
```

Reply:
It requires #5189